### PR TITLE
feat(cloud_sql): add ssl_mode to PostgreSQL example

### PR DESCRIPTION
**Pull request by feng-zhe**
## Description

Fixes #310700535

## Checklist

**Readiness**

- [X] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [ ] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [ ] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [X] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md): https://paste.googleplex.com/6576073727279104?raw

   - [X] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [X] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [X] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Comment by msampathkumar** [comment]
@feng-zhe - Thank you for sharing testing logs

**Comment by feng-zhe** [comment]
Hi @msampathkumar, thanks for reviewing the change. I saw the above 
`terraform-docs-samples-int-trigger (cloud-foundation-cicd)` failed with `--- FAIL: TestSamples/2/sqlserver_instance_private_ip (603.18s)`. I feel my change should not impact it. Please advice. Thanks.

**Comment by feng-zhe** [comment]
Thanks @msampathkumar! Somehow the CICD still fails and I cannot find anything related to this change.

**Comment by feng-zhe** [comment]
Hi @msampathkumar , any updates? Please let me know what I should to make the CICD errors go away. Thanks.

File: cloud_sql/postgres_instance_ssl_cert/main.tf [hunk 1]
```diff
@@ -22,7 +22,11 @@ resource "google_sql_database_instance" "postgres_instance" {
   settings {
```
Comment (msampathkumar) [inline cloud_sql/postgres_instance_ssl_cert/main.tf:22]: can you please update the version to `POSTGRES_14` ?

Comment (msampathkumar) [reply cloud_sql/postgres_instance_ssl_cert/main.tf:22]: Accepted version: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#database_version

**Comment by feng-zhe** [comment]
Hi @msampathkumar, just a friendly ping. Thanks.

**Comment by msampathkumar** [comment]
Error: 

```
Step #5 - "test-samples-2": TestSamples/2/sqlserver_instance_private_ip 2023-12-15T13:27:57Z command.go:185: Error: Unable to remove Service Networking Connection, err: Error waiting for Delete Service Networking Connection: Error code 9, message: Failed to delete connection; Producer services (e.g. CloudSQL, Cloud Memstore, etc.) are still using this connection.
```

**Comment by msampathkumar** [comment]
I will be doing for manual testing for approval.

**Review by msampathkumar — APPROVED** [review]
Manually tested. LGTM.

**Pull request merged**